### ISSUE TITLE
Offline / GroundDB support

### ISSUE THREAD

**Issue by yorrd** [jagi/meteor-astronomy#690]
Hey!

We're trying to combine groundDB and astronomy in our current project. However, that's kind of an issue because GroundDB doesn't expose a mongo collection but rather a `Ground.Collection`. Therefore, you're rightfully throwing this error:
![image](https://user-images.githubusercontent.com/1191486/47602859-9deb1400-d9e4-11e8-918f-830a4806457d.png)

Do you think there's any chance we can support GroundDB Collections? They implement the same interface for the most part, haven't understood why they opt to define their own type yet.

Thanks, have a great weekend!

**Comment by lukejagodzinski** [comment]
@yorrd hey! Ok I will take a look at it and maybe I will just add weak dependency on GroundDB, so it won't end up being bundled until it's being used. Doing so I could be able to check if the collection property is an instance of `Mongo.Collection` or `Ground.Collection`.

Probably there was a reason behind not extending `Mongo.Collection` so I will make sure that both works properly. However, I'm a little bit busy right now but maybe I will have some free time on the weekend. If I don't post you with an update on next Monday then please remind me to take a look at this issue.

**Comment by yorrd** [comment]
@lukejagodzinski awesome, looking forward to it!

**Comment by yorrd** [comment]
reminder if you have time :)

**Comment by lukejagodzinski** [comment]
@yorrd yep I remember about that but recently have so much private stuff going on that don't have any time to work on this but I will soon :)

**Comment by lukejagodzinski** [comment]
@yorrd ok I've finally had time to work on that. It won't be possible to use GroundDB collections in place of Mongo.Collection. There are several differences in API that I don't want to get into details right now. But it's also client only collection.

So, unfortunately Astronomy in this form can't work with GroundDB.

**Comment by yorrd** [comment]
aaaaaahhhhh shit okay. Thanks for your time though! Does it make any sense for me to have a look into it and maybe submit a PR? Or am I better off hacking together a third party offline solution?

**Comment by yorrd** [comment]
@lukejagodzinski one idea I had is manually inserting a locally stored copy of some of the db entries I need when going offline. I have seen a `simulation: boolean` option in the `.save()` callback, which prevents simulation on the client. Any chance to have an option that only simulates and doesn't send to the server? In my case this makes sense, because when starting offline, I would hydrate the local mongo db with my localstorage entries, but don't want those to be sent to the server when the user is back online because it could overwrite newer changes

**Comment by lukejagodzinski** [comment]
@yorrd if you create client only Mongo.Collection then it will save only on the client but don't know if that's what you want. If you have Mongo.Collection that is being use on both client and server then no, there is no way (at least without a lot of code rewriting) to store data only on the client

**Comment by yorrd** [comment]
I googled around a bit more and found this which works very well: https://stackoverflow.com/questions/29854322/inserting-document-into-only-client-side-of-collection 

Apparently, there's a way to access `collection._collection` to get the actual raw minimongo collection of the client. It does complain about duplicate IDs later, but I'll do some more testing on that, pretty sure it can be worked around. I wrote a 60 line custom offline solution which works like a charm except for the duplicate ID thing. 

Maybe offline isn't something to be abstracted away by something like GroundDB after all - maybe on a less automized level. We'll do some more testing with this approach internally and maybe release something flexible and astronomy-compatible to the open public.
